pkg/proc/gdbserial/rr.go

```diff
@@ -63,32 +63,42 @@ func RecordAsync(cmd []string, wd string, quiet bool, redirects [3]string) (run
 	return run, stop, nil
 }
 
-func openRedirects(redirects [3]string, quiet bool) (stdin, stdout, stderr *os.File, closefn func(), err error) {
-	toclose := []*os.File{}
+func openRedirects(redirects [3]proc.OutputRedirect, quiet bool) (stdin, stdout, stderr *os.File, closefn func(), err error) {
+	var (
+		toclose = []*os.File{}
+	)
 
-	if redirects[0] != "" {
-		stdin, err = os.Open(redirects[0])
+	if redirects[0].Path != "" {
+		stdin, err = os.Open(redirects[0].Path)
 		if err != nil {
 			return nil, nil, nil, nil, err
 		}
 		toclose = append(toclose, stdin)
-	} else {
+	} else if quiet {
```

Comment:
Can you explain this change?

Reply:
Sorry for the wrong change here. I always thought I had fixed this logic....

Reply:
done.

Reply:
You changed this line from `} else {` to `} else if quiet{`, why?